src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,410 @@
+            .attr("fill", (d, i) => { return (z[i]); });
+
+        // ===========================================================================================
+        // Set tooltip
+        // ===========================================================================================
+
+        //const tooltip = svgViewport.append("div")
+        const tooltip = d3DivContainer.append("div")
+            .attr("class", "remove")
+            .attr("id", "tooltip")
+            .attr("style", getCssToString(styles.tooltip, styles))
+            .style("position", "absolute")
+            .style("z-index", "2")
+            .style("visibility", "hidden")
+            .style("top", "4px")
+            .style("left", "12px");
+
+        let hoveredKey;
+        let hoveredValue;
+
+        // ==============================================================================differenceBy==========
+        // Set vertical line
+        // ===========================================================================================
+
+        let mouseIsOverStream = false;
+
+        // vertical line
+        const vertical = d3DivContainer.append("div")
+            //.attr("class", {style.vertical})
+            .style("position", "absolute")
+            .style("z-index", "19")
+            .style("width", "1px")
+            .style("height", `${height}px`)
+            .style("top", "10px")
+            .style("bottom", "30px")
+            .style("left", "0px")
+            .style("margin-top", `${margin.top - 10}px`)
+            .style("pointer-events", "none")
+            .style("background", "#000");
+
+        // ===========================================================================================
+        // Set legend
+        // ===========================================================================================
+
+        const legendView = d3DivContainer.append('div')
+            .attr("id", "legend")
+            .attr("style", getCssToString(styles.legend, styles))
+
+        for (let index = colorNameList.length - 1; index > 0; index--) {
+            let element = colorNameList[index];
+            let legendItemContainer = legendView.append("div")
+                .attr("style", getCssToString(styles.legendItem, styles))
+                .attr("class", "legendItem");
+
+            legendItemContainer.append("svg")
+                .attr("width", 15)
+                .attr("height", 15)
+                .style("background-color", element[1]);
+
+            legendItemContainer.append("text")
+                .attr("style", getCssToString(styles.legentItemText, styles))
+                .attr("class", "legentItemText")
+                .text(element[0]);
+        }
+
+        // ===========================================================================================
+        // Mouse events - svgViewport events
+        // 
+        // Can't split in different blocks because .on("") events 
```

Comment:
Again, this comment is here because the function is too large. Please split it into multiple ones.